packages/discovery-provider/src/challenges/cosign_challenge.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from typing import Dict
+
+from sqlalchemy.orm.session import Session
+
+from src.challenges.challenge import ChallengeManager, ChallengeUpdater
+from src.models.rewards.user_challenge import UserChallenge
+
+logger = logging.getLogger(__name__)
+
+
+class CosignChallengeUpdater(ChallengeUpdater):
+    """
+    This challenge is completed when a verified user cosigns a remix.
+    The specifier is the track ID.
+    """
+
+    def generate_specifier(self, session: Session, user_id: int, extra: Dict) -> str:
+        return f"{extra['track_id']}"
+
+    def should_create_new_challenge(
```

Comment:
wanna gate it on a blocknumber passing like this? https://github.com/AudiusProject/audius-protocol/commit/910d6c15e08709d5f823a09996887be9ec358758

(there's a bug, this logic should be in `should_create_new_challenge`.